pkg/storage/replica_range_lease.go

```diff
@@ -143,15 +142,14 @@ func (p *pendingLeaseRequest) InitOrJoinRequest(
 		// Send result of lease to all waiter channels.
 		replica.mu.Lock()
 		defer replica.mu.Unlock()
-		for i, llChan := range p.llChans {
-			// Don't send the same pErr object twice; this can lead to races. We could
-			// clone every time but it's more efficient to send pErr itself to one of
-			// the channels (the last one; if we send it earlier the race can still
-			// happen).
-			if i == len(p.llChans)-1 {
```

Comment:
Ok, disregard my comment and change this back to always cloning. Not cloning seems to cause concurrent map access. Also, in the `pErr == nil` case, I think it is clearer to send on the channel using `llChan <- err`.